packages/components/inputs/date-time-input/src/date-time-input.tsx

```diff
@@ -342,20 +388,27 @@ class DateTimeInput extends Component {
                     onMouseEnter: () => {
                       // we remove the highlight so that the user can use the
                       // arrow keys to move the cursor when hovering
+                      // @ts-ignore
                       if (isOpen) setHighlightedIndex(null);
                     },
-                    onKeyDown: (event) => {
+                    onKeyDown: (
+                      event: KeyboardEvent<HTMLButtonElement | HTMLInputElement>
```

Comment:
could we type the _event_ with `TPreventDownshiftDefaultEvent` so we don't need to cast later?

Reply:
I separated createBlurHandler and preventDownshiftDefault events. That solved the problem.